server/src/main/java/org/elasticsearch/gateway/LocalAllocateDangledIndices.java

```diff
@@ -105,6 +103,23 @@ public void messageReceived(final AllocateDangledRequest request, final Transpor
                 indexNames[i] = request.indices[i].getIndex().getName();
             }
             final String source = "allocation dangled indices " + Arrays.toString(indexNames);
+
+            var listener = new AllocationActionListener<>(ActionListener.<AllocateDangledResponse>wrap(response -> {
+                try {
+                    channel.sendResponse(new AllocateDangledResponse());
+                } catch (IOException e) {
+                    logger.warn("failed send response for allocating dangled", e);
+                }
+            }, exception -> {
+                logger.error(() -> "unexpected failure during [" + source + "]", exception);
+                try {
+                    channel.sendResponse(exception);
+                } catch (Exception inner) {
+                    inner.addSuppressed(exception);
+                    logger.warn("failed send response for allocating dangled", inner);
+                }
+            }), transportService.getThreadPool().getThreadContext());
```

Comment:
In fact, can we just do this? I don't think the logging has much value here now that this action is triggered by a REST call and therefore we can send errors back to the client.

```suggestion
            var listener = new AllocationActionListener<>(
                new ChannelActionListener<AllocateDangledResponse, AllocateDangledRequest>(channel, task.getAction(), request),
                transportService.getThreadPool().getThreadContext()
            );
```